torch/ao/quantization/fx/_decomposed.py

```diff
@@ -589,15 +589,13 @@ def quantize_per_channel(
     assert axis < input.dim(), f"Expecting axis to be < {input.dim()}"
     _quant_min_max_bounds_check(quant_min, quant_max, dtype)
     input, permute_axis_list = _permute_to_axis_zero(input, axis)
-    res = torch.zeros_like(input)
 
-    for i in range(input.size(0)):
-        res[i] = torch.clamp(
-            torch.round(input[i] * (1.0 / scales[i])) + zero_points[i],
-            quant_min,
-            quant_max,
-        )
+    new_shape = [1] * input.dim()
+    new_shape[0] = scales.shape[0]
+    scales = scales.view(new_shape)
+    zero_points = zero_points.view(new_shape)
 
+    res = torch.clamp(torch.round(input * (1.0 / scales)) + zero_points, quant_min, quant_max)
```

Comment:
does it not change how zero_points / scales need to be broadcasted?

e.g. for this new version to work, `permute` should happen to the last axis instead of `axis_zero`, rigth?

for this code to work as is, broadcast needs to happen "on the right" (which is not supported), no?

Reply:
I don't have a great understanding of PyTorch broadcasting, but as evidence that this works, I submit 1) the tests, which haven't failed yet, and 2) https://github.com/pytorch/pytorch/pull/132828 , which I linked in the summary.

Reply:
Oh, I now see in your code you're manually adding singular dims to `scale`/`zero_point`. I think a simpler way would be to move the feature-dims to the last dim (instead of zero'th dim), and then there is no need for inserting singular dims...

Reply:
> I think a simpler way would be to move the feature-dims to the last dim (instead of zero'th dim), and then there is no need for inserting singular dims as broadcasting would handle missing singular dims on the fly...

I don't understand how to do this. Please feel free to send a follow-up; it sounds easy!

Reply:
I think, this would work:
`res = torch.clamp(torch.round(input.transpose(axis, -1) * (1.0 / scales)) + zero_points, quant_min, quant_max).transpose(axis, -1)`

Also, should be noted, that both my solution and your solution change the contiguity pattern (my solution makes it `channels_last` and your solution moves the feature dim to front and makes it contig). @jerryzh168, am I correct and does it matter in this context? (or maybe `*` and `+` would propagate left-arg strides?)